cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -148,8 +148,10 @@ Status AzureOptions::ExtractFromUriQuery(const Uri& uri) {
       ARROW_ASSIGN_OR_RAISE(background_writes,
                             ::arrow::internal::ParseBoolean(kv.second));
     } else {
-      return Status::Invalid(
-          "Unexpected query parameter in Azure Blob File System URI: '", kv.first, "'");
+      // Assume these are part of a SAS token. Its not ideal to make such an assumption
+      // but given that a SAS token is a complex set of URI parameters, that could be
+      // tricky to exhaustively list I think its the best option.
+      credential_kind = CredentialKind::kSasToken;
```

Comment:
We don't have the SAS token specification that includes parameter names used by a SAS token, right?

Reply:
Yeah, I had a quick search and couldn't find what we need. If you think it's important I can try a bit harder. The closest I found seemed to be unabbreviated versions of what actually appears in the sas token.

Reply:
I found it: https://learn.microsoft.com/en-us/rest/api/storageservices/create-user-delegation-sas#construct-a-user-delegation-sas

There are many parameters but can we check them...?

Reply:
I think that doc is only for user delegated SAS tokens so I unioned it with the parameters for account and service SAS tokens and hopefully the spec is slowly changing. 

I wasn't really confident on the best way to define a constant set of strings to do `contains` checks against in C++. Since there are only 27 values, I ended up with a `constexpr` array and `std::find` but please let me know if this is not a good option.